File: Game/MonoEngine/Vertex2D.cs
namespace MonoEngine
{
    public struct Vertex2D
    {
        public Vector2 Position;
        public Vector2 TexCoord;
        public Color Color;

        public Vertex2D(Vector2 position, Vector2 texCoord, Color color)
        {
            Position = position;
            TexCoord = texCoord;
            Color = color;
        }

        public Vertex2D(Vector2 position, Vector2 texCoord) : this(position, texCoord, Color.White)
        {
        }
    }
}

File: MonoEngine/GFX/Vertex2D.cpp
#include "Vertex2D.h"

#include "../GL/VertexBuffer.h"

void Vertex2D::SetupVertexArray(const GLuint vao) {
    SetupVertexArrayFloatsAttrib(vao, 0, 0, 2, offsetof(Vertex2D, Position));
    SetupVertexArrayFloatsAttrib(vao, 1, 0, 2, offsetof(Vertex2D, TexCoord));
    SetupVertexArrayFloatsAttrib(vao, 2, 0, 4, offsetof(Vertex2D, Color));
}